tm2/pkg/p2p/transport.go

```diff
@@ -147,13 +147,19 @@ func (mt *MultiplexTransport) Close() error {
 }
 
 // Listen starts an active process of listening for incoming connections [NON-BLOCKING]
-func (mt *MultiplexTransport) Listen(addr types.NetAddress) error {
+func (mt *MultiplexTransport) Listen(addr types.NetAddress) (rerr error) {
 	// Reserve a port, and start listening
 	ln, err := net.Listen("tcp", addr.DialString())
 	if err != nil {
 		return fmt.Errorf("unable to listen on address, %w", err)
 	}
 
+	defer func() {
+		if rerr != nil {
```

Comment:
We shouldn't close the listener here, we should return the listener to queue it for closing, as we do for `bft.Node.rpcListeners` (I think you can re-use that one and rename the field to `listeners`)

Reply:
If the named return was non-nil, we definitely should be closing it, for example if `addr.Port == 0` failed to get a TCP address. The protection using the non-nil named return error ensures that we don't leak after returning.